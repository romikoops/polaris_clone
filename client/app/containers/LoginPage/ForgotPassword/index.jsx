import React from 'react'
import Formsy from 'formsy-react'
import PropTypes from '../../../prop-types'
// import { authenticationActions } from '../../actions'
import { LoadingSpinner } from '../../../components/LoadingSpinner/LoadingSpinner'
import { RoundButton } from '../../../components/RoundButton/RoundButton'
import styles from './ForgotPassword.scss'
import FormsyInput from '../../../components/FormsyInput/FormsyInput'
import getApiHost from '../../../constants/api.constants'
import { authHeader } from '../../../helpers'

const { fetch } = window

export default class ForgotPassword extends React.PureComponent {
  constructor (props) {
    super(props)
    this.state = { focus: {}, sendingEmail: false }
  }
  handleSubmit (model) {
    console.log(this.props)
    console.log(model)

    this.setState({ sendingEmail: true })

    const payload = {
      ...model,
      redirect_url: getApiHost() // TBD - + 'path_to_password_reset'
    }

    fetch(`${getApiHost()}/auth/password`, {
      method: 'POST',
      headers: { ...authHeader(), 'Content-Type': 'application/json' },
      body: JSON.stringify(payload)
    }).then((promise) => {
      promise.json().then((response) => {
        // TBD - render some animation instead of reloading the page
        window.location.replace('/')
      })
    })
  }
  handleInvalidSubmit () {
    if (!this.state.submitAttempted) this.setState({ submitAttempted: true })
  }

  handleFocus (e) {
    this.setState({
      focus: {
        ...this.state.focus,
        [e.target.name]: e.type === 'focus'
      }
    })
  }

  render () {
    const { focusStyles, theme } = this.props
    const { sendingEmail } = this.state
    return (
      <Formsy
        className={styles.forgot_password_form}
        name="form"
        onValidSubmit={model => this.handleSubmit(model)}
        onInvalidSubmit={model => this.handleInvalidSubmit(model)}
      >
        <div className="form-group">
          <label htmlFor="email">Email</label>
          <FormsyInput
            type="text"
            className={styles.form_control}
            onFocus={e => this.handleFocus(e)}
            onBlur={e => this.handleFocus(e)}
            name="email"
            placeholder="enter your email"
            submitAttempted={this.state.submitAttempted}
            validationErrors={{ isDefaultRequiredValue: 'Must not be blank' }}
            required
          />
          <hr style={this.state.focus.email ? focusStyles : {}} />
        </div>
        <p style={{ fontSize: '13px', textAlign: 'justify', width: '271px' }}>
          By clicking {'\'Send\''}, a link will be sent to your
          email address, in order to reset your password.
        </p>
        <div className={`${styles.form_group_submit_btn} layout-row layout-align-center-center`}>
          <RoundButton text="Send" theme={theme} size="small" active />
          <div className={styles.spinner}>{sendingEmail && <LoadingSpinner />}</div>
        </div>
      </Formsy>
    )
  }
}

ForgotPassword.propTypes = {
  theme: PropTypes.theme,
  focusStyles: PropTypes.objectOf(PropTypes.any)
}

ForgotPassword.defaultProps = {
  theme: null,
  focusStyles: {}
}
