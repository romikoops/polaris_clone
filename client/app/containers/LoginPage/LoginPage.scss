@import "../../styles/defaults";
:local {
	form.login_form {
		width: 317px;

		::-webkit-input-placeholder { /* Chrome/Opera/Safari */
			color: $grey;
			font-size: 14px;
		}
		::-moz-placeholder { /* Firefox 19+ */
			color: $grey;
			font-size: 14px;
		}
		:-ms-input-placeholder { /* IE 10+ */
			color: $grey;
			font-size: 14px;
		}
		:-moz-placeholder { /* Firefox 18- */
			color: $grey;
			font-size: 14px;
		}
		input, label {
			display: block;
		}
		hr {
			border: .5px solid rgb(10, 10, 10);
			margin: 6px 0 24px 0;
		}
		.form_group_submit_btn {
			padding-top: 25px;
		}
		label {
			text-transform: uppercase;
			margin-bottom: 4px;
		}
	}


	.form_control {
		font-size: 14px;
		background: transparent;
		border: none;
		border-radius: 4px;
		width: 100%;
		padding: 4px 0;

		&:focus {
			outline: none;
			background: rgba(black, .016);
		}
		&:focus + hr {
			border: 1px solid $primary;
			margin-bottom: 23px;
		}
	}


	.forget_password_link {
		text-decoration: none;
		color: inherit;
		font-size: 13px;
		float: right;
		margin-top: -20px;	
	}
}