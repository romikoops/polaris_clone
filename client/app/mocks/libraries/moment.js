import moment from 'moment'

require('moment-timezone')

moment.tz.setDefault('Europe/Amsterdam')
