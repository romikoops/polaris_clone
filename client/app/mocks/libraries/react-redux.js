import * as ReactRedux from 'react-redux'

ReactRedux.connect = jest.fn().mockImplementation((mapStateToProps, mapDispatchToProps) => Component => Component)
