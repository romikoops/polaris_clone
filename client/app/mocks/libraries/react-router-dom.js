import * as ReactRouterDom from 'react-router-dom'

ReactRouterDom.withRouter = jest.fn().mockImplementation(x => x)
