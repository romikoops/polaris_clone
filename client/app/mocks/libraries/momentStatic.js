import moment from 'moment'

require('moment-timezone')

const constantDate = new Date('June 13, 2017 04:41:20 GMT+1:00')

// eslint-disable-next-line
Date = class extends Date {
  constructor () {
    super()

    return constantDate
  }
}

moment.tz.setDefault('Europe/Amsterdam')
