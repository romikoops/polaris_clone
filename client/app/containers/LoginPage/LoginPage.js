import React from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import { authenticationActions } from '../../actions';
import { RoundButton } from '../../components/RoundButton/RoundButton';
import { Alert } from '../../components/Alert/Alert';
import styles from './LoginPage.scss';
import Formsy from 'formsy-react';
import FormsyInput from '../../components/FormsyInput/FormsyInput';

class LoginPage extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            submitAttempted: false,
            focus: {},
            alertVisible: false
        };

        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleInvalidSubmit = this.handleInvalidSubmit.bind(this);
        this.handleFocus = this.handleFocus.bind(this);
        this.hideAlert = this.hideAlert.bind(this);
    }
    componentWillMount() {
        if (this.props.loginAttempt && !this.state.alertVisible) {
            this.setState({ alertVisible: true });
        }
    }

    componentWillReceiveProps(nextProps) {
        if (nextProps.loginAttempt && !this.state.alertVisible) {
            this.setState({ alertVisible: true });
        }
    }

    hideAlert() {
        this.setState({ alertVisible: false });
    }

    handleSubmit(model) {
        const { username, password } = model;
        const { dispatch, req, noRedirect } = this.props;
        if (username && password) {
            dispatch(authenticationActions.login({
                email: username,
                password: password,
                shipmentReq: req,
                noRedirect
            }));
        }
    }
    handleInvalidSubmit() {
        if (!this.state.submitAttempted) this.setState({ submitAttempted: true });
    }

    handleFocus(e) {
        this.setState({
            focus: {
                ...this.state.focus,
                [e.target.name]: e.type === 'focus'
            }
        });
    }

    render() {
        const { loggingIn, theme } = this.props;
        const focusStyles = {
            borderColor: theme && theme.colors ? theme.colors.primary : 'black',
            borderWidth: '1.5px',
            borderRadius: '2px',
            margin: '-0.75px 0 28.25px 0'
        };
        const alert = this.state.alertVisible ? (
            <Alert
                message={{ type: 'error', text: 'Wrong username or password' }}
                onClose={this.hideAlert}
                timeout={10000}
            />
        ) : '';
        return (
            <Formsy
                className={styles.login_form}
                name="form"
                onValidSubmit={this.handleSubmit}
                onInvalidSubmit={this.handleInvalidSubmit}
            >
                { alert }
                <div className="form-group">
                    <label htmlFor="username">Username</label>
                    <FormsyInput
                        type="text"
                        value={this.props.user ? this.props.user.email : ''}
                        className={styles.form_control}
                        onFocus={this.handleFocus}
                        onBlur={this.handleFocus}
                        name="username"
                        placeholder="enter your username"
                        submitAttempted={this.state.submitAttempted}
                        validationErrors={{isDefaultRequiredValue: 'Must not be blank'}}
                        required
                    />
                    <hr style={this.state.focus.username ? focusStyles : {}}/>
                </div>
                <div className="form-group">
                    <label htmlFor="password">Password</label>
                    <FormsyInput
                        type="password"
                        value={this.props.user ? this.props.user.password : ''}
                        className={styles.form_control}
                        name="password"
                        placeholder="enter your password"
                        submitAttempted={this.state.submitAttempted}
                        validationErrors={{isDefaultRequiredValue: 'Must not be blank'}}
                        required
                    />
                    <hr style={this.state.focus.password ? focusStyles : {}}/>
                    <a href="#" className={styles.forget_password_link}>forgot password?</a>
                </div>
                <div className={`form-group ${styles.form_group_submit_btn}`}>
                    <RoundButton text="Sign In" theme={theme} active/>

                    <div style={{height: '10px'}}>
                        {loggingIn &&
                            <img src="data:image/gif;base64,R0lGODlhEAAQAPIAAP///wAAAMLCwkJCQgAAAGJiYoKCgpKSkiH/C05FVFNDQVBFMi4wAwEAAAAh/hpDcmVhdGVkIHdpdGggYWpheGxvYWQuaW5mbwAh+QQJCgAAACwAAAAAEAAQAAADMwi63P4wyklrE2MIOggZnAdOmGYJRbExwroUmcG2LmDEwnHQLVsYOd2mBzkYDAdKa+dIAAAh+QQJCgAAACwAAAAAEAAQAAADNAi63P5OjCEgG4QMu7DmikRxQlFUYDEZIGBMRVsaqHwctXXf7WEYB4Ag1xjihkMZsiUkKhIAIfkECQoAAAAsAAAAABAAEAAAAzYIujIjK8pByJDMlFYvBoVjHA70GU7xSUJhmKtwHPAKzLO9HMaoKwJZ7Rf8AYPDDzKpZBqfvwQAIfkECQoAAAAsAAAAABAAEAAAAzMIumIlK8oyhpHsnFZfhYumCYUhDAQxRIdhHBGqRoKw0R8DYlJd8z0fMDgsGo/IpHI5TAAAIfkECQoAAAAsAAAAABAAEAAAAzIIunInK0rnZBTwGPNMgQwmdsNgXGJUlIWEuR5oWUIpz8pAEAMe6TwfwyYsGo/IpFKSAAAh+QQJCgAAACwAAAAAEAAQAAADMwi6IMKQORfjdOe82p4wGccc4CEuQradylesojEMBgsUc2G7sDX3lQGBMLAJibufbSlKAAAh+QQJCgAAACwAAAAAEAAQAAADMgi63P7wCRHZnFVdmgHu2nFwlWCI3WGc3TSWhUFGxTAUkGCbtgENBMJAEJsxgMLWzpEAACH5BAkKAAAALAAAAAAQABAAAAMyCLrc/jDKSatlQtScKdceCAjDII7HcQ4EMTCpyrCuUBjCYRgHVtqlAiB1YhiCnlsRkAAAOwAAAAAAAAAAAA==" />
                        }
                    </div>
                </div>
            </Formsy>
        );
    }
}

function mapStateToProps(state) {
    const { loggingIn, loginAttempt, user } = state.authentication;
    return {
        loggingIn,
        loginAttempt,
        user
    };
}

LoginPage.propTypes = {
    dispatch: PropTypes.func,
    loggingIn: PropTypes.any,
    theme: PropTypes.object
};

const connectedLoginPage = connect(mapStateToProps)(LoginPage);
export { connectedLoginPage as LoginPage };
