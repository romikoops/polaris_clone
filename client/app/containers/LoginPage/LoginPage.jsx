/* eslint-disable jsx-a11y/label-has-associated-control */
import React from 'react'
import { connect } from 'react-redux'
import Formsy from 'formsy-react'
import { get } from 'lodash'
import { withNamespaces } from 'react-i18next'
import { authenticationActions } from '../../actions'
import { RoundButton } from '../../components/RoundButton/RoundButton'
import { LoadingSpinner } from '../../components/LoadingSpinner/LoadingSpinner'
import styles from './LoginPage.scss'
import FormsyInput from '../../components/FormsyInput/FormsyInput'
import ForgotPassword from './ForgotPassword'

class LoginPage extends React.Component {
  static redirectToSamlLogin () {
    window.location.href = '/saml/init'
  }

  constructor (props) {
    super(props)

    this.state = {
      submitAttempted: false,
      focus: {},
      showLoginForm: !props.authMethods.includes('saml')
    }

    this.handleSubmit = this.handleSubmit.bind(this)
    this.handleInvalidSubmit = this.handleInvalidSubmit.bind(this)
    this.handleFocus = this.handleFocus.bind(this)
  }

  handleSubmit (model) {
    const { email, password } = model
    const {
      dispatch, req, noRedirect, redirectUrl
    } = this.props
    dispatch(authenticationActions.login({
      email,
      password,
      req,
      noRedirect,
      redirectUrl
    }))
  }

  handleInvalidSubmit () {
    if (!this.state.submitAttempted) this.setState({ submitAttempted: true })
  }

  handleFocus (e) {
    this.setState({
      focus: {
        ...this.state.focus,
        [e.target.name]: e.type === 'focus'
      }
    })
  }

  toggleShowLoginForm () {
    this.setState((prevState) => ({ showLoginForm: !prevState.showLoginForm }))
  }

  renderForgotPassword () {
    this.setState({ forgotPassword: true })
  }

  render () {
    const { loggingIn, theme, scope, authMethods, t } = this.props
    const focusStyles = {
      borderColor: theme && theme.colors ? theme.colors.primary : 'black',
      borderWidth: '1.5px',
      borderRadius: '2px',
      margin: '-1px 0 29px 0'
    }
    const allowForgotPassword = !get(scope, 'user_restrictions.profile.password')

    if (this.state.forgotPassword) {
      return <ForgotPassword focusStyles={focusStyles} theme={theme} />
    }

    const ie11Positioning =
      navigator.userAgent.includes('MSIE') || document.documentMode ? styles.login_ie_11 : ''
    const hasSamlLogin = authMethods.includes('saml')
    const { focus, submitAttempted, showLoginForm } = this.state
    const { email, password } = focus

    return (
      <div>
        {hasSamlLogin && (
          <div className={`form-group ${showLoginForm ? '' : styles.samlPadding}`}>
            <RoundButton
              handleNext={() => LoginPage.redirectToSamlLogin()}
              classNames="ccb_saml_signin"
              text={t('login:samlText', { text: scope.saml_text })}
              theme={theme}
              active
            />
            { showLoginForm && <hr className={styles.saml_border} /> }
          </div>
        )}
        <Formsy
          className={`${styles.login_form} ${ie11Positioning}`}
          name="form"
          onValidSubmit={this.handleSubmit}
          onInvalidSubmit={this.handleInvalidSubmit}
        >
          <div className={`form-group ${styles.form} ${showLoginForm ? '' : styles.hideForm}`}>
            <label htmlFor="email">{t('login:email')}</label>
            <FormsyInput
              type="text"
              className={styles.form_control}
              onFocus={this.handleFocus}
              onBlur={this.handleFocus}
              name="email"
              placeholder={t('login:enterEmail')}
              submitAttempted={submitAttempted}
              validationErrors={{ isDefaultRequiredValue: t('login:mustNotBeBlank') }}
              required
            />
            <hr style={email ? focusStyles : {}} />
          </div>
          <div className={`form-group ${styles.form} ${showLoginForm ? '' : styles.hideForm}`}>
            <label htmlFor="password">{t('login:password')}</label>
            <FormsyInput
              type="password"
              className={styles.form_control}
              name="password"
              placeholder={t('login:enterPassword')}
              submitAttempted={submitAttempted}
              validationErrors={{ isDefaultRequiredValue: t('login:mustNotBeBlank') }}
              required
            />
            <hr style={password ? focusStyles : {}} />
            {allowForgotPassword && (
              <a
                onClick={() => this.renderForgotPassword()}
                className={`forgotPassword ${styles.forget_password_link}`}
              >
                {t('login:forgotPassword')}
              </a>
            )}
          </div>
          <div
            className={`form-group ${styles.form_group_submit_btn}
            ${styles.form} ${showLoginForm ? '' : styles.hideForm}`}
          >
            <RoundButton classNames="ccb_signin" text={t('login:signIn')} theme={theme} active />
            <div className={styles.spinner}>{loggingIn && <LoadingSpinner />}</div>
          </div>
        </Formsy>
        <div>
          {(hasSamlLogin && !showLoginForm) && (
            <a onClick={() => this.toggleShowLoginForm()} className={`showLogin ${styles.admin_login_text}`}>
              {t('login:emailLogin')}
            </a>
          )}
        </div>
      </div>
    )
  }
}

function mapStateToProps (state) {
  const { authentication, app } = state
  const {
    loggingIn, loginAttempt, user, noRedirect, req, redirectUrl
  } = authentication
  const { scope, auth_methods: authMethods } = app.tenant

  return {
    loggingIn,
    loginAttempt,
    user,
    noRedirect,
    req,
    redirectUrl,
    scope,
    authMethods
  }
}

LoginPage.defaultProps = {
  loggingIn: false,
  theme: null,
  loginAttempt: false,
  noRedirect: false,
  req: null,
  authMethods: []
}

const connectedLoginPage = connect(mapStateToProps)(LoginPage)
const translatedLoginPage = withNamespaces(['login'])(connectedLoginPage)
export { translatedLoginPage as LoginPage }

export default translatedLoginPage
