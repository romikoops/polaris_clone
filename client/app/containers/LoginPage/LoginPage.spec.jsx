import '../../mocks/libraries/react-redux'
import * as React from 'react'
import { shallow } from 'enzyme'
import {
  user, tenant
} from '../../mocks/index'

import LoginPage from './LoginPage'

const propsBase = {
  logginIn: false,
  loginAttempt: false,
  user,
  noRedirect: false,
  req: null,
  scope: tenant.scope,
  authMethods: []
}

test('shallow render', () => {
  expect(shallow(<LoginPage {...propsBase} />)).toMatchSnapshot()
})

test('shallow render with saml login', () => {
  const props = {
    ...propsBase,
    authMethods: ['saml']
  }
  expect(shallow(<LoginPage {...props} />)).toMatchSnapshot()
})

test('stoggleShowLogin sets showLoginForm to true', () => {
  const props = {
    ...propsBase,
    authMethods: ['saml']
  }
  const wrapper = shallow(<LoginPage {...props} />)
  const instance = wrapper.instance()
  expect(instance.state.showLoginForm).toBe(false)
  instance.toggleShowLoginForm()
  expect(instance.state.showLoginForm).toBe(true)
})
test('clicking showLogin button sets showLoginForm to true', () => {
  const props = {
    ...propsBase,
    authMethods: ['saml']
  }
  const wrapper = shallow(<LoginPage {...props} />)
  const instance = wrapper.instance()
  expect(instance.state.showLoginForm).toBe(false)
  wrapper.find('.showLogin').simulate('click')
  expect(instance.state.showLoginForm).toBe(true)
})
test('clicking forgotPassword button sets forgotPassword to true', () => {
  const wrapper = shallow(<LoginPage {...propsBase} />)
  const instance = wrapper.instance()
  expect(instance.state.forgotPassword).toBe(undefined)
  expect(instance.state.showLoginForm).toBe(true)
  wrapper.find('.forgotPassword').simulate('click')
  expect(instance.state.forgotPassword).toBe(true)
})
