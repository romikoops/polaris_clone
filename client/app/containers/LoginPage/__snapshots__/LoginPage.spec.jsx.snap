// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow render 1`] = `
<div>
  <Formsy
    className="login_form "
    disabled={false}
    getErrorMessage={[Function]}
    getErrorMessages={[Function]}
    getValue={[Function]}
    hasValue={[Function]}
    isFormDisabled={[Function]}
    isFormSubmitted={[Function]}
    isPristine={[Function]}
    isRequired={[Function]}
    isValid={[Function]}
    isValidValue={[Function]}
    mapping={null}
    name="form"
    onChange={[Function]}
    onError={[Function]}
    onInvalid={[Function]}
    onInvalidSubmit={[Function]}
    onReset={[Function]}
    onSubmit={[Function]}
    onValid={[Function]}
    onValidSubmit={[Function]}
    preventExternalInvalidation={false}
    resetValue={[Function]}
    setValidations={[Function]}
    setValue={[Function]}
    showError={[Function]}
    showRequired={[Function]}
    validationErrors={null}
  >
    <div
      className="form-group form "
    >
      <label
        htmlFor="email"
      >
        Email
      </label>
      <Formsy(FormsyInput)
        className="form_control"
        innerRef={[Function]}
        name="email"
        onBlur={[Function]}
        onFocus={[Function]}
        placeholder="Enter your email"
        required={true}
        submitAttempted={false}
        type="text"
        validationError=""
        validationErrors={
          Object {
            "isDefaultRequiredValue": "Must not be blank",
          }
        }
        validations={null}
      />
      <hr
        style={Object {}}
      />
    </div>
    <div
      className="form-group form "
    >
      <label
        htmlFor="password"
      >
        Password
      </label>
      <Formsy(FormsyInput)
        className="form_control"
        innerRef={[Function]}
        name="password"
        placeholder="Enter your password"
        required={true}
        submitAttempted={false}
        type="password"
        validationError=""
        validationErrors={
          Object {
            "isDefaultRequiredValue": "Must not be blank",
          }
        }
        validations={null}
      />
      <hr
        style={Object {}}
      />
      <a
        className="forgotPassword forget_password_link"
        onClick={[Function]}
      >
        forgot password?
      </a>
    </div>
    <div
      className="form-group form_group_submit_btn
            form "
    >
      <RoundButton
        active={true}
        back={false}
        classNames="ccb_signin"
        disabled={false}
        flexContainer="100"
        handleDisabled={null}
        handleNext={null}
        icon=""
        iconClass=""
        inverse={false}
        size=""
        text="Sign In"
        theme={null}
      />
      <div
        className="spinner"
      />
    </div>
  </Formsy>
  <div />
</div>
`;

exports[`shallow render with saml login 1`] = `
<div>
  <div
    className="form-group samlPadding"
  >
    <RoundButton
      active={true}
      back={false}
      classNames="ccb_saml_signin"
      disabled={false}
      flexContainer="100"
      handleDisabled={null}
      handleNext={[Function]}
      icon=""
      iconClass=""
      inverse={false}
      size=""
      text="Sign in with {{text}}"
      theme={null}
    />
  </div>
  <Formsy
    className="login_form "
    disabled={false}
    getErrorMessage={[Function]}
    getErrorMessages={[Function]}
    getValue={[Function]}
    hasValue={[Function]}
    isFormDisabled={[Function]}
    isFormSubmitted={[Function]}
    isPristine={[Function]}
    isRequired={[Function]}
    isValid={[Function]}
    isValidValue={[Function]}
    mapping={null}
    name="form"
    onChange={[Function]}
    onError={[Function]}
    onInvalid={[Function]}
    onInvalidSubmit={[Function]}
    onReset={[Function]}
    onSubmit={[Function]}
    onValid={[Function]}
    onValidSubmit={[Function]}
    preventExternalInvalidation={false}
    resetValue={[Function]}
    setValidations={[Function]}
    setValue={[Function]}
    showError={[Function]}
    showRequired={[Function]}
    validationErrors={null}
  >
    <div
      className="form-group form hideForm"
    >
      <label
        htmlFor="email"
      >
        Email
      </label>
      <Formsy(FormsyInput)
        className="form_control"
        innerRef={[Function]}
        name="email"
        onBlur={[Function]}
        onFocus={[Function]}
        placeholder="Enter your email"
        required={true}
        submitAttempted={false}
        type="text"
        validationError=""
        validationErrors={
          Object {
            "isDefaultRequiredValue": "Must not be blank",
          }
        }
        validations={null}
      />
      <hr
        style={Object {}}
      />
    </div>
    <div
      className="form-group form hideForm"
    >
      <label
        htmlFor="password"
      >
        Password
      </label>
      <Formsy(FormsyInput)
        className="form_control"
        innerRef={[Function]}
        name="password"
        placeholder="Enter your password"
        required={true}
        submitAttempted={false}
        type="password"
        validationError=""
        validationErrors={
          Object {
            "isDefaultRequiredValue": "Must not be blank",
          }
        }
        validations={null}
      />
      <hr
        style={Object {}}
      />
      <a
        className="forgotPassword forget_password_link"
        onClick={[Function]}
      >
        forgot password?
      </a>
    </div>
    <div
      className="form-group form_group_submit_btn
            form hideForm"
    >
      <RoundButton
        active={true}
        back={false}
        classNames="ccb_signin"
        disabled={false}
        flexContainer="100"
        handleDisabled={null}
        handleNext={null}
        icon=""
        iconClass=""
        inverse={false}
        size=""
        text="Sign In"
        theme={null}
      />
      <div
        className="spinner"
      />
    </div>
  </Formsy>
  <div>
    <a
      className="showLogin admin_login_text"
      onClick={[Function]}
    >
      Login with email and password
    </a>
  </div>
</div>
`;
